crates/docs/src/validation.rs

```diff
@@ -1,60 +1,54 @@
-use regex::Regex;
-use std::{
-    env, fs, io,
-    path::{Path, PathBuf},
-};
+use std::{fs, io, path::Path};
 
-const EXTENSION: Option<&str> = Some("md");
-
-pub struct Snippet {
-    pub command: String,
-    pub file_path: String,
-    pub line_start: usize,
-}
+use crate::snippet::{Snippet, SnippetConfig, SnippetType};
 
-impl Snippet {
-    pub fn to_command_args(&self) -> Vec<String> {
-        let cleaned_command = self
-            .command
-            .lines()
-            .map(str::trim_end)
-            .collect::<Vec<&str>>()
-            .join(" ")
-            .replace(" \\", "");
-
-        shell_words::split(&cleaned_command)
-            .expect("Failed to parse snippet string")
-            .into_iter()
-            .map(|arg| arg.trim().to_string())
-            .collect()
-    }
-}
+const EXTENSION: Option<&str> = Some("md");
 
-pub fn extract_snippets_from_file(file_path: &Path, re: &Regex) -> io::Result<Vec<Snippet>> {
+pub fn extract_snippets_from_file(
+    file_path: &Path,
+    snippet_type: &SnippetType,
+) -> io::Result<Vec<Snippet>> {
     let content = fs::read_to_string(file_path)?;
     let file_path_str = file_path
         .to_str()
         .expect("Failed to get file path")
         .to_string();
 
-    let snippets = re
+    let snippets = snippet_type
+        .get_re()
         .captures_iter(&content)
         .filter_map(|caps| {
-            let command_match = caps.get(1)?;
             let match_start = caps.get(0)?.start();
+            let config_str = caps
+                .get(1)
+                .map_or_else(String::new, |m| m.as_str().to_string());
+            let command_match = caps.get(2)?;
+            let output = caps.get(3).map(|m| m.as_str().to_string());
+
```

Comment:
We have everything very nicely abstracted except for this part. You have to know the internal structure of the regex here to know from which capture to get which part

Reply:
Could we at least use named captures for this?

https://docs.rs/regex/latest/regex/struct.Captures.html#method.name